src/composer.ts

```diff
@@ -57,6 +62,27 @@ export enum AtomicTransactionComposerStatus {
   COMMITTED,
 }
 
+function populateForeignArray<Type>(
```

Comment:
nit: would be great to add comment to explain what this function is doing, i.e., calculate the object index for foreign object array...
btw, it seems the name is somewhat misleading tho?

Reply:
It's based on the name for this function from goal: https://github.com/algorand/go-algorand/blob/29a10dad66d0ed5b793a18967e8ee234788c834f/cmd/goal/application.go#L1091